packages/cli-repl/src/cli-repl.ts

```diff
@@ -795,6 +798,76 @@ export class CliRepl implements MongoshIOProvider {
     }
   }
 
+  verifyPlatformSupport(): void {
+    if (this.cliOptions.quiet) {
+      return;
+    }
+
+    // Typings for process.getReport haven't been updated
+    // (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40140)
+    const processReport = process.report?.getReport() as unknown as
+      | {
+          header: {
+            glibcVersionRuntime: string;
+          };
+        }
+      | undefined;
+    if (!processReport) {
+      return;
+    }
+
+    const warnings: string[] = [];
+    const RECOMMENDED_GLIBC = '>=2.28.0';
+    const RECOMMENDED_OPENSSL = '>=3.0.0';
+    const RECOMMENDED_NODEJS = '>=20.0.0';
```

Comment:
I was not sure if there is a better place to put these constants in so I kept them here.